File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int binarySearch(vector<int>& nums, int target, int l, int r) {

        if (l > r) {
            return -1;
        }

        int m = l + (r - l) / 2;

        if (nums[m] == target) {
            return m;
        }

        if (nums[m] < target) {
            return binarySearch(nums, target, m + 1, r);
        }

        return binarySearch(nums, target, l, m - 1);
    }

    int search(vector<int>& nums, int target) {
        return binarySearch(nums, target, 0, nums.size() - 1);
    }
    
};

File: Data Structures & Algorithms/search-insert-position/submission-0.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:

        for i in range(0, len(nums)-1):
            if (nums[i] < target and nums[i+1] > target):
                nums.insert(i+1, target)
                return i+1
            elif(target > nums[len(nums)-1]):
                nums.insert(len(nums), target)
                return len(nums)-1
            elif(target < nums[0]):
                nums.insert(0, target)
                return 0
        
        

        


        